pkgs/development/python-modules/starlette/default.nix

```diff
@@ -20,12 +20,20 @@
 
 buildPythonPackage rec {
   pname = "starlette";
-  version = "0.13.0";
+
+  # This is not the latest version of Starlette, however, later
+  # versions of Starlette break FastAPI due to
+  # https://github.com/tiangolo/fastapi/issues/683. Please update when
+  # possible. FastAPI is currently Starlette's only dependent.
+
+  version = "0.12.9";
   disabled = isPy27;
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "6bd414152d40d000ccbf6aa40ed89718b40868366a0f69fb83034f416303acef";
+  src = fetchFromGitHub {
```

Comment:
Why is these needed?

Reply:
There was some discussion about [this here](https://github.com/NixOS/nixpkgs/pull/48633#issuecomment-432732045) and other links that I can't find now. A standard Python distribution doesn't have the test suite as part of the distribution. Ideally, Python distributions would have some standard for self testing, but they don't.

Reply:
Ah, ok for the tests. Thank you.